### PR TITLE
print: Print single map values

### LINKED ISSUE #2779: print() single map values
```
BEGIN
{
  @x[1] = 1;
  print(@x[1])
}
```
This script errors out with: `The map passed to print() should not be indexed by a key`.

`printf` is able to print a single map value - `print` should be able to as well.

### PR DESCRIPTION
**Pull request by arthurshau**
Allows indexing a map with key(s) to print single map values while using the print() function.

Resolves #2779.

##### Checklist

- [ ] Language changes are updated in `man/adoc/bpftrace.adoc`
- [x] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [x] The new behaviour is covered by tests

### CODE REVIEW

File: src/ast/passes/semantic_analyser.cpp [hunk 1]
```diff
@@ -930,9 +932,10 @@ void SemanticAnalyser::visit(Call &call)
       if (arg.is_map) {
         Map &map = static_cast<Map &>(arg);
         if (map.vargs != nullptr) {
-          LOG(ERROR, call.loc, err_)
-              << "The map passed to " << call.func << "() should not be "
-              << "indexed by a key";
+          if (call.vargs->size() != 1)
+            LOG(ERROR, call.loc, err_)
+                << "Single-value (i.e. indexed) map print cannot take top "
+                   "or div arguments.";
```
Comment (viktormalik) [inline src/ast/passes/semantic_analyser.cpp:938]: I don't think that the err message should be top- and div-specific. We may add more cases requiring multiple `print` arguments in future and would have to always update the message:
```suggestion
                << "Single-value (i.e. indexed) map print cannot take additional "
                   "arguments.";
```

**Review by viktormalik — COMMENTED** [review]
LGTM, thanks!

Just one comment about an error message.

File: src/ast/passes/semantic_analyser.cpp (hunk 1 above)
Comment (arthurshau) [reply src/ast/passes/semantic_analyser.cpp:938]: Done, thanks!

**Comment by arthurshau** [comment]
Addressed comments

File: src/ast/passes/semantic_analyser.cpp (hunk 1 above)
Comment (viktormalik) [reply src/ast/passes/semantic_analyser.cpp:938]: It seems that you changed the wrong error message :-)

Comment (arthurshau) [reply src/ast/passes/semantic_analyser.cpp]: Whoops...should probably get to bed

**Review by viktormalik — APPROVED** [review]
Thanks!

**Review by ajor — APPROVED** [review]
Nice!

**Pull request merged**